s3/src/main/java/io/ddf/s3/S3DDFManager.java

```diff
@@ -269,4 +297,17 @@ public S3DataSourceCredentials getCredential() {
   public void stop() {
     // TODO: Does s3 connection has to be closed?
   }
+
+  /**
+   * @brief Get the bucket and path out of a given uri.
+   * @param path
+   * @return
+   */
+  public static List<String> getBucketAndKey(String path) throws DDFException {
+    int firstSlash = path.indexOf('/');
+    if (firstSlash == -1 || firstSlash == path.length()) {
+      throw new DDFException(String.format("The path %s is not a valid s3 path", path));
```

Comment:
@PangZhi this means `bucket-name` or `bucket-name/` is not a valid path? What if I put my files at the root of a bucket?